services/meta/ota.md

```diff
@@ -7,34 +7,44 @@ This service is used to manage over-the-air (OTA) upgrades of end devices.
 
 #### Interfaces
 
-Type | Interface               | Value type | Description
------|-------------------------|------------|--------------
-out  | evt.ota_progress.report | int_map    | Shows upgrade progress and estimated time left.
-out  | evt.ota_end.report      | object     | Sent on upgrade end with upgrade status.
+Type | Interface               | Value type | Description                                                                                                      |
+-----|-------------------------|------------|------------------------------------------------------------------------------------------------------------------|
+in   | cmd.status.get_report   | null       | Requests update progress information.                                                                            |
+out  | evt.status.report       | object     | Shows update progress and estimated time left. See [`ota_report_object`](#definitions) for detailed information. |
+in   | cmd.update.install      | string     | Starts an update of the device. Value should be the full path to the update file.                                | 
 
+## Definitions
+
+* `ota_report_object` is an object with the following structure:
+
+| Field          | Type   | Example         | Description                                                                                                     |
+|----------------|--------|-----------------|-----------------------------------------------------------------------------------------------------------------|
+| state          | string | `"IN PROGRESS"` | One of the `state` values.                                                                                      |
+| substate       | string | `"TRANSFERING"` | One of the `substate` values.                                                                                   |       
+| progress       | int    | `10`            | Optional, only available in substate `TRANSFERING`. Progress percentage of the update transfer.                 |
+| remaining_min  | int    | `3`             | Optional, only available in substate `TRANSFERING`. Remaining time until the end of the transfer (minutes).     | 
+| remaining_sec  | int    | `35`            | Optional, only available in substate `TRANSFERING`. Remaining time until the end of the transfer (seconds).     |
+| last_update    | int    | `132351314`     | Optional. Unix timestamp of the last started update process.                                                    |        
```

Comment:
Suggestion of possible improvement:
* What about string type and RFC3339?
* Wouldn't it make sense to rename the field and add: "started_at" and "finished_at".

Each object report represents a certain stage in the process, which:
* might have never started (both started_at and finished_at will be skipped)
* started but did not finish
* started and finished

Reply:
@BudzikM @eternalamenity @PaulinaDrozda @Brzezoo

Reply:
RFC3339 -> yes,

started / finished -> yes

Reply:
I need more time, I will review all changes soon.

> in general we - the app, have to add support for the current/old `ota` service now. It's not the first time in which we are encountering smaller or bigger breaking changes, and imo we need a way to provide a way to easily differentiate them. Right now we have to read all possibilities (current and previous, and previous are not present in the documentation) and add special logic that will define which "version" to choose. One of possibilities that I see it to keep `ota` service as it is right now, and introduce `ota_v2`. that would add clear line between them

Ota_v2 will have all different topics, so will not be confused with v1.  You want to specify cleary documentation version or you are asking how to check and recognize ota version in practice ? Sending ota.status.get_report might be a way.

Reply:
@BudzikM It's more like @GenixPL is requesting that both old and new interfaces are kept in the documentation in a visible way so at any time in the future frontend can easily check old versions to maintain backwards compatibility while refactoring the app.

* When (or if) we create a FIMP v2 it will be a separate set of docs with everything reworked which is at least 1.5-2 years away from production. So if we create V2, entire current state of documents will be copied and visible in previous version section in a real time. It might even be possible we will be maintaining and updating FIMPv1 and FIMPv2 at the same time for some possibly long period as well.

* We are not introducing breaking changes to the existing V1 version of FIMP with following expectations:
** something was never used in production or supported, then we can simply remove it
** something was ill-implemented and we need to introduce breaking changes to fix it
** something was used internally and we can handle a three-stage update

So...

Reply:
@PaulinaDrozda This is what I suggest to do:

Keep the old interfaces in the documentation and mark them as deprecated.

We pledge ourselves to keep them in documentation V1 for the eternity.

Going for a Kitchen Knife now to make a blood pledge. @GenixPL happy?